src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/GlobalSearchService.kt

```diff
@@ -61,6 +66,103 @@ class GlobalSearchService(
     }
   }
 
+  data class NomisContext(var nomisFinished: Boolean, var num: String?)
+
+  fun compareIndex(): Pair<List<String>, List<String>> {
+    val onlyInIndex = mutableListOf<String>()
+    val onlyInNomis = mutableListOf<String>()
+
+    val allNomis =
+      prisonerIndexService.getAllNomisOffenders(0, Int.MAX_VALUE).offenderIds!!.sortedBy { it.offenderNumber }
```

Comment:
This is pretty hefty, I thought Id see if it breaks dev before getting sophisticated about streaming etc.